open_dread_rando/door_patcher.py

```diff
@@ -320,4 +333,59 @@ def update_minimap_for_shield(self, shield: Container, shield_type: ActorData, d
     def remove_all_shields(self):
         for scenario in ["s010_cave", "s020_magma", "s030_baselab", "s040_aqua", "s050_forest", "s060_quarantine", "s070_basesanc", "s080_shipyard", "s090_skybase"]:
             bmmap = self.editor.get_scenario_map(scenario)
-            bmmap.raw.Root.mapBlockages = Container()
\ No newline at end of file
+            bmmap.raw.Root.mapBlockages = Container()
+    
+    def rename_all_shields(self):
+        for scenario in ["s010_cave", "s020_magma", "s030_baselab", "s040_aqua", "s050_forest", "s060_quarantine", "s070_basesanc", "s080_shipyard", "s090_skybase"]:
+            brfld = self.editor.get_scenario(scenario)
+            shielded_doors = []
+            for layer_name, actor_name, actor in list(brfld.all_actors()):
```

Comment:
Making a copy before iterating is really only if your for loop will add/remove an actor. Otherwise it's preferable to not do so.

Reply:
This was done because it was trying to access the shield actors that had been renamed, and then having issues finding them. I'll add a comment to this line to clarify that.